internal/executor/executor.go

```diff
@@ -528,6 +530,11 @@ func stepExecAfter(req *ExecutionRequest) bool {
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
 
+	fakeRequest := &ExecutionRequest{
+		Arguments: args,
+	}
+	cmd.Env = buildEnv(fakeRequest)
```

Comment:
I can see why you did this, and I might well have done the same - however I think it is better to change the function signature of `buildEnv` to just take a `map[string]string` instead. That function doens't need anything else in `req`. Would you mind updating this PR to do that instead, and then you don't need `fakeRequest` anymore. 

You'll need to update `stepExec` to pass `req.Arguments` as well, instead of just `req`. 

Is that OK? It makes keeps things a little cleaner.

Reply:
Sure, that makes sense. Will fix

I actually thought about it, but I was wary of refactoring (might not see the Grand Plan ™️  or whatnot...)